### ISSUE TITLE
Responsive Settings in PrimeFaces Grid

### ISSUE THREAD

**Issue by JoelProminic** [prominic/Moonshine-IDE#321]
While reviewing #266 today, I noticed that the generated code was using the responsive grid CSS oddly:

IMAGE

```
        <div class="ui-g">
          <div class="ui-g-6 ui-lg-6 ui-sm-3 ui-md-3 ui-xl-6">
            <div style="width:100%;height:100%;" class="flexHorizontalLayout flexHorizontalLayoutLeft flexHorizontalLayoutTop">
              <p:outputLabel style="width:100px;height:30px;" value="R1C1"/>
            </div>
          </div>
          <div class="ui-g-6 ui-lg-6 ui-sm-3 ui-md-3 ui-xl-6">
            <div style="width:100%;height:100%;" class="flexHorizontalLayout flexHorizontalLayoutLeft flexHorizontalLayoutTop">
              <p:outputLabel style="width:100px;height:30px;" value="R1C2"/>
            </div>
          </div>
        </div>
        <div class="ui-g">
          <div class="ui-g-6 ui-lg-6 ui-sm-3 ui-md-3 ui-xl-6">
            <div style="width:100%;height:100%;" class="flexHorizontalLayout flexHorizontalLayoutLeft flexHorizontalLayoutTop">
              <p:outputLabel style="width:100px;height:30px;" value="R2C1"/>
            </div>
          </div>
          <div class="ui-g-6 ui-lg-6 ui-sm-3 ui-md-3 ui-xl-6">
            <div style="width:100%;height:100%;" class="flexHorizontalLayout flexHorizontalLayoutLeft flexHorizontalLayoutTop">
              <p:outputLabel style="width:100px;height:30px;" value="R2C2"/>
            </div>
          </div>
        </div>
```

"ui-g-6 ui-lg-6 ui-sm-3 ui-md-3 ui-xl-6" will cause a cell to use half of the grid width for large screens, but only one quarter of the width on small screens/windows, which is the opposite of what we want.

If we don't want to support responsive behavior for the Grid, we should just use "ui-g-6" (or the appropriate "ui-6-*" option for the number of columns).   As an advanced option, we could allow the user to specify the responsive configuration from the properties.

For more details, see page 252 of the [documentation](https://www.primefaces.org/docs/guide/primefaces_user_guide_6_2.pdf)

You can also see examples [here](https://www.primefaces.org/showcase/ui/panel/grid.xhtml)

**Comment by piotrzarzycki21** [comment]
Well I'm trying to provide the most suitable values, but I think it should be responsibility of user what kind of classes should be added. What do you think to add options where there will be possibility to add ?
ui-g-* - some value
ui-sm-* - some value

I'm fine also to have always class ui-g-6 - I the end we will have to adjust layout for the final application.

**Comment by JoelProminic** [comment]
Sorry, @piotrzarzycki21, I missed your response...

For now, I don't see an option to customize the ui-* properties in the Grid component.  In this case, I would suggest that we only use  "ui-g-X", so that the grid is consistent across all platforms.

If we add support for the other "ui-*-X" classes in the futures, I think I would still default them to use the same value as "ui-g", and allow the users to customize the responsive behavior from this starting point.

**Comment by piotrzarzycki21** [comment]
I think I have resolved that one as part of #333

**Comment by JoelProminic** [comment]
The "ui-" classes all match for now.  We will need to revisit this after we complete #333.

          <div class="ui-g-5 ui-lg-5 ui-sm-5 ui-md-5 ui-xl-5">
